wgsl/index.bs

```diff
@@ -279,17 +279,27 @@ The events are ordered due to:
 
 ## Kinds of errors ## {#kinds-of-errors}
 
-A <dfn export>program error</dfn> is a failure to satisfy the requirements of this specification.
+A WebGPU implementation may fail to process a shader for two reasons:
 
-There are three kinds of errors, corresponding to the shader lifecycle:
+* A <dfn export>program error</dfn> occurs if the shader does not satisfy the requirements of the [SHORTNAME] or WebGPU specifications.
+* An <dfn noexport>unexpected error</dfn> may occur
+    even when all [SHORTNAME] and WebGPU requirements have been satisfied.
+    Possible causes include:
+    * The shaders are too complex, exceeding the capabilities of the implementation,
+        but in a way not easily captured by prescribed [[#limits|limits]].
+        Simplifying the shaders may work around the issue.
+    * A defect in the WebGPU implementation.
+
+
+Errors are organized into three kinds, corresponding to the shader lifecycle:
```

Comment:
This feels oddly worded because above we appear to have organized errors into two kinds. Are the errors a cross product of these two?

Maybe this would be better phrased along the lines of:

Errors may occur at three different points in the shader lifecycle:
* Shader-creation time: this corresponds to an error occurring during a call to createShaderModule API method
* ...

Reply:
Yeah, I struggled with this. 
I need it to tie into the statement later:


> Each requirement in this specification corresponds to a single kind of error.  

But maybe the trick is to reword *that* more clearly.

Reply:
PTAL at the latest commit